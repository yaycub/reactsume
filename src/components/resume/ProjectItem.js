import React from 'react';
import PropTypes from 'prop-types';

const ProjectItem = ({ title, tech = '', description }) => {
  return (
    <section>
      <h4>{title}</h4>
      <p>{tech}</p>
      <p>{description}</p>
    </section>
  );
};

ProjectItem.propTypes = {
  title: PropTypes.string.isRequired,
  tech: PropTypes.string,
  description: PropTypes.string.isRequired
};

export default ProjectItem;
