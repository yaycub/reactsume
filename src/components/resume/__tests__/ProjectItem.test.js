import React from 'react';
import { shallow } from 'enzyme';
import ProjectItem from '../ProjectItem';

describe('ProjectItem component', () => {
  it('can render a ProjectItem', () => {
    const wrapper = shallow(<ProjectItem title="my title" tech="some tech" description="this is a project" />);
    expect(wrapper).toMatchSnapshot();
  });
});
