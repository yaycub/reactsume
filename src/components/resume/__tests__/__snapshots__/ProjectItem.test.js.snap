// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem component can render a ProjectItem 1`] = `
<section>
  <h4>
    my title
  </h4>
  <p>
    some tech
  </p>
  <p>
    this is a project
  </p>
</section>
`;
